matminer/featurizers/function.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import division
+
+"""
+This module includes code to featurize data based on
+common mathematical expressions used as features.
+
+Note that the approach here has opted for a sympy-based
+parsing of string expressions, rather than explicit
+python functions.  The primary reason this has been
+done is to provide for better support for book-keeping
+(e. g. with feature labels), substitution, and elimination
+of symbolic redundancy, which sympy is well-suited for.
+"""
+
+import numpy as np
+from sympy.parsing.sympy_parser import parse_expr
+import sympy as sp
+import itertools
+from six import string_types
+
+from collections import OrderedDict
+
+from matminer.featurizers.base import BaseFeaturizer
+
+
+# Default expressions to include in function featurizer
+default_exps = ["x", "1/x", "sqrt(x)", "1/sqrt(x)", "x**2", "x**-2", "x**3",
+                "x**-3", "log(x)", "1/log(x)", "exp(x)", "exp(-x)"]
+
+
+# TODO: feature labels might be latexified
+# TODO: parallelization of substitution, think this might be
```

Comment:
I agree, parallelization over rows is already implemented and probably preferred.